source/code/plugin/in_kube_podinventory.rb

```diff
@@ -183,55 +290,80 @@ def parse_and_emit_records(podInventory, serviceList)
+              records.push(record.dup)
+
+              #Generate ContainerInventory records for windows nodes so that we can get image and image tag in property panel
+              if sendWindowsContainerInventoryRecord == true
+                containerInventoryRecord = populateWindowsContainerInventoryRecord(container, record, containerEnvVariableHash, batchTime)
+                containerInventoryRecords.push(containerInventoryRecord)
+              end
+            end
           else # for unscheduled pods there are no status.containerStatuses, in this case we still want the pod
-            records.push(record) 
+            records.push(record)
           end  #container status block end
           records.each do |record|
             if !record.nil?
-              record['PodRestartCount'] = podRestartCount
+              record["PodRestartCount"] = podRestartCount
               wrapper = {
-                          "DataType"=>"KUBE_POD_INVENTORY_BLOB",
-                          "IPName"=>"ContainerInsights",
-                          "DataItems"=>[record.each{|k,v| record[k]=v}]
+                          "DataType" => "KUBE_POD_INVENTORY_BLOB",
+                          "IPName" => "ContainerInsights",
+                          "DataItems" => [record.each { |k, v| record[k] = v }],
                         }
               eventStream.add(emitTime, wrapper) if wrapper
-            end  
-          end  
+            end
+          end
+          # Send container inventory records for containers on windows nodes
+          winContainerCount += containerInventoryRecords.length
+          containerInventoryRecords.each do |cirecord|
+            if !cirecord.nil?
+              ciwrapper = {
+                "DataType" => "CONTAINER_INVENTORY_BLOB",
+                "IPName" => "ContainerInsights",
+                "DataItems" => [cirecord.each { |k, v| cirecord[k] = v }],
+              }
+              eventStream.add(emitTime, ciwrapper) if ciwrapper
+            end
+          end
         end  #podInventory block end
+
         router.emit_stream(@tag, eventStream) if eventStream
         router.emit_stream(@@MDMKubePodInventoryTag, eventStream) if eventStream
         if telemetryFlush == true
-          ApplicationInsightsUtility.sendHeartBeatEvent("KubePodInventory")
-          ApplicationInsightsUtility.sendMetricTelemetry("PodCount", podInventory['items'].length , {})
-          ApplicationInsightsUtility.sendMetricTelemetry("ControllerCount", controllerSet.length , {})
+          telemetryProperties = {}
+          telemetryProperties["Computer"] = @@hostName
+          ApplicationInsightsUtility.sendCustomEvent("KubePodInventoryHeartBeatEvent", telemetryProperties)
+          ApplicationInsightsUtility.sendMetricTelemetry("PodCount", podInventory["items"].length, {})
+          ApplicationInsightsUtility.sendMetricTelemetry("ControllerCount", controllerSet.length, {})
+          if winContainerCount > 0
```

Comment:
get rid of this check

Reply:
Double checked this. Since I am sending ContainerInventoryHeartbeatEvent in CAdvisorMetricsApiClient, leaving this as is since this is just information to know if we did send any container inventory record.